src/shared/utils.js

```diff
@@ -1,92 +1,77 @@
 /**
- * Generate an empty 2d array.
+ * Generate an empty "virtual" 2d array.
  *
- * @param {Number} boardSize - Array dimensions.
+ * @param {Number} boardSize - Array dimensions (will be squared).
  *
- * @return {Array[]}
+ * @return {Array}
  */
 export function generateEmptyBoard( boardSize ) {
 	const array = [];
 
-	for ( let i = 0; i < boardSize; i++ ) {
-		array[ i ] = [];
-
-		for ( let j = 0; j < boardSize; j++ ) {
-			array[ i ][ j ] = null;
-		}
+	for ( let i = 0; i < Math.pow( boardSize, 2 ); i++ ) {
+		array[ i ] = null;
 	}
 
 	return array;
 }
 
 /**
- * Get a winner from a board.
+ * Extract rows, columns and diagonals as arrays from a virtual 2d array.
  *
- * @param {Array[]} board - Board as a 2d array.
+ * @param {Array} board - Board as a virtual 2d array.
  *
- * @return {String|null} - Winner value or `null`.
+ * @return { rows: Array, cols: Array, diags: Array };
  */
-export function getWinner( board ) {
-	// Check rows & cols.
-	for ( let i = 0; i < board.length; i++ ) {
-		const row = board[ i ];
+export function extractBoardParts( board ) {
+	const size = parseInt( Math.sqrt( board.length ) );
 
-		if ( arrayHasEqualValues( row ) ) {
-			return row[ 0 ];
-		}
+	const rows = [];
+	const cols = [];
+	const diags = [ [], [] ];
 
-		const col = getColumn( board, i );
+	for ( let i = 0; i < board.length; i += size ) {
+		const index = i / size;
 
-		if ( arrayHasEqualValues( col ) ) {
-			return col[ 0 ];
-		}
-	}
-
-	// Check LTR diagonal.
-	const diagLTR = getDiagonal( board, 'ltr' );
+		cols[ index ] = [];
+		rows[ index ] = [];
 
-	if ( arrayHasEqualValues( diagLTR ) ) {
-		return diagLTR[ 0 ];
-	}
+		// Calculate rows / cols
+		for ( let j = 0; j < size; j++ ) {
+			rows[ index ].push( board[ i + j ] );
+			cols[ index ].push( board[ i / 3 + j * size ] );
+		}
 
-	// Check RTL diagonal.
-	const diagRTL = getDiagonal( board, 'rtl' );
+		// Calculate diagonals.
+		const iteration = i / 3;
```

Comment:
Move the iteration variable above the rows/col loop and use it in line 41.

Reply:
Done